drools-ansible-rulebook-integration-api/src/main/java/org/drools/ansible/rulebook/integration/api/rulesengine/SessionStats.java

```diff
@@ -198,6 +203,11 @@ public long getMaxAvailableMemory() {
         return MemoryMonitorUtil.getMaxAvailableMemory();
     }
 
+    public long getPeakMemory() {
+        // a little verbose but it is more accurate because peakMemory is not updated in all cases
+        return Math.max(peakMemory, MemoryMonitorUtil.getUsedMemory());
+    }
```

Comment:
As `peakMemory` is updated on every 64 processEvent calls, there might be a case where a returned SessionStats has higher `usedMemory` than `peakMemory`. To avoid unnecessary doubt by users, do one more update. (e.g. on Json marshalling)